homeassistant/util/__init__.py

```diff
@@ -116,6 +121,38 @@ def get_random_string(length=10):
     return ''.join(generator.choice(source_chars) for _ in range(length))
 
 
+def load_json(filename: str) -> Union[List, Dict]:
+    """Load JSON data from a file and return as dict.
+
+    Defaults to returning empty dict (if e.g. file not found).
+    """
+    try:
+        with open(filename, encoding='utf-8') as fdesc:
+            return json.loads(fdesc.read())
+    except FileNotFoundError:
+        _LOGGER.debug('JSON file not found: %s', filename)
+    except ValueError:
+        _LOGGER.warning('Could not parse JSON content: %s', filename)
+    except OSError as error:
+        _LOGGER.warning('JSON file reading failed: %s (%s)',
+                        filename, error)
+    return {}  # (also evaluates to False)
+
+
+def save_json(filename: str, config: Union[List, Dict]):
+    """Save JSON data to a file.
+
+    Returns True on success, else None.
+    """
+    try:
+        with open(filename, 'w', encoding='utf-8') as fdesc:
+            fdesc.write(json.dumps(config, sort_keys=True, indent=4))
+            return True
+    except OSError as error:
```

Comment:
Same here, should we blow up to make sure that both user and code knows it did not manage to save anything.